File: matlab/ch03/compID.m
function ind = compID(cname) 
% Assignment of identification number 
cname = upper(cname);  
switch cname    
case {'FLUORINE','F2'}, ind = 1;                
case {'CHLORINE','CL2'}, ind = 2;    
case {'SULFUR DIOXIDE','SO2'}, ind = 3;       
case {'CARBON MONOXIDE','CO'}, ind = 4;  
case {'CARBON DIOXIDE','CO2'}, ind = 5;       
case {'HYDROGEN CHLORIDE','HCL'}, ind = 6;     
case {'AMMONIA','NH3'}, ind = 7;                   
case {'WATER','H2O'}, ind = 8;     
case {'HYDROGEN PEROXIDE','H2O2'}, ind = 9; 
case {'HYDROGEN','H2'}, ind = 10;    
case {'NITROGEN','N2'}, ind = 11;                 
case {'OXYGEN','O2'}, ind = 12;    
case {'ETHYLENE','C2H4'}, ind = 13;             
case {'METHANE','CH4'}, ind = 14;    
case {'ETHANE','C2H6'}, ind = 15;                
case {'PROPANE','C3H8'}, ind = 16;    
case {'BENZENE','C6H6'}, ind = 17;              
case {'TOLUENE','C7H8'}, ind = 18;    
case {'ANILINE','C6H7N'}, ind = 19;            
case {'PHENOL','C6H6O'}, ind = 20;    
case {'CYCLOPROPANE','C3H6'}, ind = 21;     
case {'CYCLOHEXANE','C6H12'}, ind = 22;     
case {'1,3 BUTADIENE','C4H6'}, ind = 23;    
case {'METHANOL','CH4O'}, ind = 24;    
case {'CHLOROFORM','CHCL3'}, ind = 25;    
case {'CARBON TETRACHLORIDE','CCL4'}, ind = 26; 
end 

File: python/ch03/compID.py
def compID(cname):
    """
    화합물 이름을 입력받아 해당 ID 번호를 반환하는 함수
    """
    # 입력값을 대문자로 변환
    cname = cname.upper()
    
    # 화합물과 ID 매핑 딕셔너리
    comp_map = {
        'FLUORINE': 1, 'F2': 1,
        'CHLORINE': 2, 'CL2': 2,
        'SULFUR DIOXIDE': 3, 'SO2': 3,
        'CARBON MONOXIDE': 4, 'CO': 4,
        'CARBON DIOXIDE': 5, 'CO2': 5,
        'HYDROGEN CHLORIDE': 6, 'HCL': 6,
        'AMMONIA': 7, 'NH3': 7,
        'WATER': 8, 'H2O': 8,
        'HYDROGEN PEROXIDE': 9, 'H2O2': 9,
        'HYDROGEN': 10, 'H2': 10,
        'NITROGEN': 11, 'N2': 11,
        'OXYGEN': 12, 'O2': 12,
        'ETHYLENE': 13, 'C2H4': 13,
        'METHANE': 14, 'CH4': 14,
        'ETHANE': 15, 'C2H6': 15,
        'PROPANE': 16, 'C3H8': 16,
        'BENZENE': 17, 'C6H6': 17,
        'TOLUENE': 18, 'C7H8': 18,
        'ANILINE': 19, 'C6H7N': 19,
        'PHENOL': 20, 'C6H6O': 20,
        'CYCLOPROPANE': 21, 'C3H6': 21,
        'CYCLOHEXANE': 22, 'C6H12': 22,
        '1,3 BUTADIENE': 23, 'C4H6': 23,
        'METHANOL': 24, 'CH4O': 24,
        'CHLOROFORM': 25, 'CHCL3': 25,
        'CARBON TETRACHLORIDE': 26, 'CCL4': 26
    }
    
    # 해당 화합물이 딕셔너리에 있으면 ID 반환, 없으면 None 반환
    return comp_map.get(cname)

# 사용 예시:
# print(compID('CO2'))  # 출력: 5